Source/WebCore/dom/Element.cpp

```diff
@@ -205,8 +205,9 @@ static bool shouldAutofocus(const Element& element)
         return false;
 
     auto& document = element.document();
-    if (!element.isConnected() || !document.hasBrowsingContext())
+    if (!element.isInShadowTree() || !element.isConnected() || !document.hasBrowsingContext())
```

Comment:
Clearly, this should read `element.isInShadowTree()` instead! That's why tests are failing.

Reply:
Thought about it more, and we should write `if (!element.isInDocumentTree() || !document.hasBrowsingContext())`

Reply:
Yeah, I've considered that but perhaps people not familiar with shadow DOM wouldn't know being inside a document tree is different from being connected to a document.

Reply:
Sure, but writing it out as `isInShadowTree() || !element.isConnected()` doesn’t make that any better in my opinion. I personally think it’s easier to explain "you need to use `isInDocumentTree` instead of `isConnected` here" is actually easier to explain than "also check if it’s in a shadow tree".

Reply:
Okay, that's a fair point. Will update.